multimodal/src/autogluon/multimodal/utils/data.py

```diff
@@ -576,7 +576,14 @@ def infer_dtypes_by_model_names(model_config: DictConfig):
     return allowable_dtypes, fallback_dtype
 
 
-def split_train_tuning_data(train_data, tuning_data, holdout_frac, is_classification, label_column, seed):
+def split_train_tuning_data(
+    train_data: Union[pd.DataFrame, str],
+    tuning_data: Optional[Union[pd.DataFrame, str]] = None,
+    holdout_frac: Optional[float] = None,
+    is_classification: Optional[bool] = False,
+    label_column: Optional[str] = None,
+    seed: Optional[int] = 123,
+):
```

Comment:
Data loading should not be done during data splitting. This is not the logical place for it.

Reply:
I have considered loading data in `predictor.py` before calling `split_train_tuning_data`, but I need repeat the same code of loading data in `matcher.py`, which increase the boilerplate code. How about changing this function name to `prepare_train_tuning_data` to make it more than just splitting data?

Reply:
Considering PR #3004, move it outside of `split_train_tuning_data`.

Reply:
@Innixma Any further questions on this?

Reply:
Since it is moved outside, I am ok to merge.

The ideal solution long term is using a mix-in or having both predictor.py and matcher.py inherit from the same abstract class. This would avoid code dupe.